cmd/process-agent/collector.go

```diff
@@ -176,7 +176,15 @@ func (l *Collector) runCheck(c checks.Check, results *api.WeightedQueue) {
 	} else {
 		checks.StoreCheckOutput(c.Name(), nil)
 	}
-	l.messagesToResults(start, c.Name(), messages, results)
+
+	if c.Name() == config.PodCheckName {
+		l.messagesToResults(start, config.PodCheckMetadataName, messages[:len(messages)/2], results)
```

Comment:
> if we now deploy the agent with the configuration of manifestCollection Disabled would it mean we collect the data as before (metadata+manifest) together? If not wasn't that the goal?
This line is what confuses me, you only take half of the payload, does it also contain the yaml as before?

Reply:
This PR doesn’t change the current payload content, it only allows agent to send the a copy of Yaml to our `orchmanif` endpoint as well.

The ff `IsManifestCollectionEnabled ` can be confusing indeed. No matter `IsManifestCollectionEnabled` is true or false, the agent always collect manifest payload. https://github.com/DataDog/datadog-agent/blob/45bf8f76a82f435e337128969d832d40b36889ee/pkg/collector/corechecks/cluster/orchestrator/processors/processor.go#L180-L184

But we only send manifest payload when `IsManifestCollectionEnabled` is true :
cluster agent : 
https://github.com/DataDog/datadog-agent/blob/21c37a41f8747126e6839e38ab7de77acaac7478/pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go#L154-L157

process agent:
https://github.com/DataDog/datadog-agent/blob/ef3dc770d5f4ba82247a91941a2f86b38de58400/cmd/process-agent/collector.go#L180-L184


So back to your question here, I just append manifest payload after our current payload,  the struct of the array 
`messages` should look like this :
```
{
current payload A (metadata+manifest),
current payload B (metadata+manifest),
current payload C (metadata+manifest),
new manifest payload A (manifest),
new manifest payload B (manifest),
new manifest payload C (manifest),
}
```

Reply:
Oh that makes sense. Thanks for the clarification. That is a great explanation. You should update your PR to reflect that information.